packages/module/src/FieldBuilder/FieldBuilder.tsx

```diff
@@ -0,0 +1,330 @@
+          cells.push(<div key="empty-placeholder" />);
+        }
+      }
+
+      // Determine span based on number of children
+      const cellSpan = cells.length === 1 ? 10 : 5;
+
+      return (
+        <Grid 
+          key={`field-row-${index}`} 
+          hasGutter 
+          className="pf-v6-u-mb-md"
+          role="group"
+          aria-labelledby={rowLabelId}
+        >
+          {/* Visually hidden but accessible label for this row group */}
+          <div id={rowLabelId} className="pf-v6-screen-reader">
+            {rowGroupLabelPrefix} {rowNumber}
+          </div>
+          
+          {/* Map over the user's components and wrap each one in a GridItem with dynamic spans. */}
+          {cells.map((cell, cellIndex) => (
+            <GridItem key={cellIndex} span={cellSpan}>
+              {cell}
+            </GridItem>
+          ))}
+          {/* Automatically add the remove button as the last item in the row. */}
+          <GridItem span={2}>
+            <Button
+              ref={createRemoveButtonRef(index)}
+              variant="plain"
+              aria-label={`Remove ${rowGroupLabelPrefix.toLowerCase()} ${rowNumber}`}
+              onClick={() => handleRemoveRow(index)}
+              icon={<MinusCircleIcon />}
+            />
+          </GridItem>
+        </Grid>
+      );
+    });
+  };
+
+  return (
+    // The entire component is wrapped in a FormGroup to be semantically correct.
+    <FormGroup {...formGroupProps}>
+      <Flex direction={{ default: 'column' }} spaceItems={{ default: 'spaceItemsNone' }}>
+        {/* ARIA Live Region for announcing dynamic changes */}
+        <div 
+          ref={liveRegionRef}
+          className="pf-v6-screen-reader"
+          aria-live="polite" 
+          aria-atomic="true"
+          role="status"
+        />
+
+        {/* Render the column labels */}
+        <Grid hasGutter className="pf-v6-u-mb-md">
+          <GridItem span={secondColumnLabel ? 5 : 10}>
+            <span id={firstColumnLabelId} className="pf-v6-c-form__label-text">
+              {firstColumnLabel}
+            </span>
+          </GridItem>
+          {secondColumnLabel && (
+            <GridItem span={5}>
+              <span id={secondColumnLabelId} className="pf-v6-c-form__label-text">
+                {secondColumnLabel}
+              </span>
+            </GridItem>
+          )}
+          {/* Empty GridItem to align with the remove button column */}
+          {/* <GridItem span={2} /> */}
+        </Grid>
+
+        {/* Render all the dynamic rows of fields */}
+        {renderRows()}
+
+        {/* The "Add" button for creating a new row */}
+        <FlexItem className="pf-v6-u-mt-md">
+          <Button 
+            ref={addButtonRef}
+            variant="link" 
+            isInline 
+            onClick={handleAddRow} 
+            icon={<PlusCircleIcon />} 
+            {...addButtonProps}
+          >
+            {addButtonProps.children || 'Add another'}
```

Comment:
Nit: we could add a "addButtonContent" prop instead of having to pass `addButtonProps.children` inside an object.